block/slvalidator.go

```diff
@@ -170,6 +170,31 @@ func (v *SettlementValidator) ValidateDaBlocks(slBatch *settlement.ResultRetriev
 		if err != nil {
 			return err
 		}
+
+		// we compare the sequencer address between SL state info and DA block
+		// if next sequencer is not set, we check if the sequencer hash is equal to the next sequencer hash
+		// because it did not change. If the next sequencer is set, we check if the next sequencer hash is equal on the
+		// last block of the batch
+		isLastBlock := i == len(slBatch.BlockDescriptors)-1
+		if slBatch.NextSequencer != "" && isLastBlock {
+			nextSequencer := v.blockManager.State.Sequencers.GetByAddress(slBatch.NextSequencer)
```

Comment:
what if we don't have the nextSequencer in the store? we should probably fetch all the sequencers again (we can't count on polling to be exact in terms of syncing). we can use https://github.com/dymensionxyz/dymint/pull/1163/files#diff-c5773c2f4ace59a21cd36af8a0c95b46135151ea34f2b4bc09e6cbecc0c3e46aR169